File: lista1/l1z1/w01b/app/src/main/java/com/example/w01b/MainActivity.java
package com.example.w01b;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private int r1;
    private int r2;
    private int score;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        score = 0;
        roll();
    }

    private void roll() {
        TextView textView = (TextView) findViewById(R.id.Points);
        textView.setText("Punkciki: " + score);
        Random random = new Random();
        r1 = random.nextInt(10);
        r2 = random.nextInt(10);
        TextView leftClick = (TextView) findViewById(R.id.LeftClick);
        leftClick.setText("" + r1);
        TextView rightClick = (TextView) findViewById(R.id.RightClick);
        rightClick.setText("" + r2);
    }

    public void BuuttonLeft(View view) {
        if(r1 >= r2) {
            score++;
            Toast.makeText(this, "Dobrze", Toast.LENGTH_SHORT).show();
        }
        else {
            score--;
            Toast.makeText(this, "Źle", Toast.LENGTH_SHORT).show();
        }
        roll();
    }

    public void BuuttonRight(View view) {
        if(r1 <= r2) {
            score++;
            Toast.makeText(this, "Dobrze", Toast.LENGTH_SHORT).show();
        }
        else {
            score--;
            Toast.makeText(this, "Źle", Toast.LENGTH_SHORT).show();
        }
        roll();
    }
}

File: lista2/l2z1/zadanko1/app/src/main/java/com/example/zadanko1/MainActivity.kt
package com.example.zadanko1

import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import android.os.CountDownTimer
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity(), OnCellClickListener{
    lateinit var gridRecyclerView : RecyclerView
    lateinit var mineGridRecyclerAdapter : MineGridRecyclerAdapter
    lateinit var game : MinesweeperGame
    lateinit var smiley: TextView
    lateinit var timer: TextView
    lateinit var flag: TextView
    lateinit var flagsCount: TextView
    lateinit var countDownTimer: CountDownTimer
    var secondsElapsed: Int = 0
    var timerStarted: Boolean = false


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        flag = findViewById(R.id.activity_main_flag)
        flagsCount = findViewById(R.id.activity_main_flagsleft)
        flag.setOnClickListener {
            game.toggleMode()
            if (game.isFlagMode()) {
                val border: GradientDrawable = GradientDrawable()
                border.setColor(-0x1)
                border.setStroke(1, -0x1000000)
                flag.background = border
            } else {
                val border = GradientDrawable()
                border.setColor(-0x1)
                flag.background = border
            }
        }

        smiley = findViewById(R.id.activity_main_smiley)
        smiley.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                println("Smiley klik")
                game = MinesweeperGame(9,9)
                mineGridRecyclerAdapter.setCells(game.getMineGrid().getCells())
                timerStarted = false
                countDownTimer.cancel()
                secondsElapsed = 0
                timer.setText(R.string.default_count)
                flagsCount.setText(String.format("%03d", game.getNumberOfBombs() - game.getFlagCount()))

            }
        })

        timer = findViewById(R.id.activity_main_timer)
        timerStarted = false
        countDownTimer = object : CountDownTimer(999000L, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                secondsElapsed += 1
                timer.text = kotlin.String.format("%03d", secondsElapsed)
            }

            override fun onFinish() {
                game.outOfTime()
                Toast.makeText(applicationContext, "Game Over: Timer Expired", Toast.LENGTH_SHORT)
                    .show()
                game.getMineGrid().revealAllBombs()
                mineGridRecyclerAdapter.setCells(game.getMineGrid().getCells())
            }
        }
        gridRecyclerView = findViewById(R.id.activity_main_grid)
        gridRecyclerView.layoutManager = GridLayoutManager(this, 9)
        game = MinesweeperGame(9, 9)
        mineGridRecyclerAdapter = MineGridRecyclerAdapter(game.getMineGrid().getCells(),this)
        gridRecyclerView.adapter = mineGridRecyclerAdapter
        flagsCount.setText(String.format("%03d", game.getNumberOfBombs() - game.getFlagCount()))
    }

    override fun onCellClick(cell: Cell) {
        game.handleCellClick(cell)

        flagsCount.setText(String.format("%03d", game.getNumberOfBombs() - game.getFlagCount()))
        if(!timerStarted){
            countDownTimer.start()
            timerStarted = true
        }
        if(game.isGameOver()){
            Toast.makeText(applicationContext, "Game is Over", Toast.LENGTH_SHORT).show()
            game.getMineGrid().revealAllBombs()
        }
        if (game.isGameWon()) {
            //countDownTimer.cancel();
            Toast.makeText(applicationContext, "Game Won", Toast.LENGTH_SHORT).show();
            game.getMineGrid().revealAllBombs();
        }
        mineGridRecyclerAdapter.setCells(game.getMineGrid().getCells())

    }
}